src/utils.js

```diff
@@ -210,18 +210,15 @@ export const delay = duration =>
 export const getTimeOfDay = () => {
   const now = new Date();
   const hourOfDay = now.getHours();
-
-  if (hourOfDay <= 4) {
-    return 'night';
-  } else if (hourOfDay <= 11) {
-    return 'morning';
+  let dayType = 'night';
+  if (hourOfDay <= 11) {
+    dayType = 'morning';
   } else if (hourOfDay <= 17) {
-    return 'afternoon';
+    dayType = 'afternoon';
   } else if (hourOfDay <= 21) {
-    return 'evening';
-  } else {
-    return 'night';
-  }
+    dayType = 'evening';
+  } 
+  return dayType;
 };
 
```

Comment:
The old version has the benefit that the logic can be read and understood from top to bottom. I think readability is the most important factor in functions like this that aren't performance critical, so I'd rather leave it like it is.

Reply:
@karlsander @joshwcomeau 
I feel like a function must have a one way out i.e one return statement 
I.e There's a single place you have to look to trace back and figure out what a function returns. I'm happy to keep it back.

Reply:
IMO it reads a lot easier when each conditional directly `return`s the result instead of storing it in a varible until the end of the function.

Reply:
> I feel like a function must have a one way out i.e one return statement
I.e There's a single place you have to look to trace back and figure out what a function returns. I'm happy to keep it back.

Yeah, I can see certain cases where that'd be beneficial, but I think more often I prefer functions to return as early as possible. This way, it can be read from top-down, and when you find the case that matches, you can stop reading it.

Especially in cases like:

```
const getUserData = (user) => {
  if (!user || !user.data) {
    return null;
  }

  // actual core logic here
}
```

Early returns are my best friend for avoiding unnecessary nesting and letting readers short-circuit if they don't actually need to see the entire body of logic.

But yeah, that's a bit of a tangent. In this case, I'd prefer if it was changed back.

Reply:
Got you! I will keep that back. Thanks for the discussion on it.